observatory/results/tabs.js

```diff
@@ -0,0 +1,92 @@
+import { html } from "lit-html";
+import { Scoring } from "./scoring";
+import { Csp } from "./csp";
+import { Cookies } from "./cookies";
+import { RawHeaders } from "./raw_headers";
+import { History } from "./history";
+import { Comparison } from "./comparison";
+
+/**
+ * @typedef {import('../constants').ObservatoryResult} ObservatoryResult
+ * @typedef {import('lit').TemplateResult} TemplateResult
+ */
+
+/**
+ * @param {{
+ *   result: ObservatoryResult,
+ *   selectedTab: number,
+ *   onTabSelect: (index: number, key: string) => void
+ * }} props
+ * @returns {TemplateResult}
+ */
+export function Tabs({ result, selectedTab = 0, onTabSelect }) {
+  const tabs = [
+    {
+      label: "Scoring",
+      key: "scoring",
+      element: html`${Scoring({ result })}`,
+    },
+    {
+      label: "CSP analysis",
+      key: "csp",
+      element: html`${Csp({ result })}`,
+    },
+    {
+      label: "Cookies",
+      key: "cookies",
+      element: html`${Cookies({ result })}`,
+    },
+    {
+      label: "Raw server headers",
+      key: "headers",
+      element: html`${RawHeaders({ result })}`,
+    },
+    {
+      label: "Scan history",
+      key: "history",
+      element: html`${History({ result })}`,
+    },
+    {
+      label: "Benchmark comparison",
+      key: "benchmark",
+      element: html`${Comparison({ result })}`,
+    },
+  ];
+
+  return html`
+    <section class="scan-results">
+      <h2 class="result" id="scan-results-header">Scan results</h2>
+      <ol
+        class="tabs-list"
+        role="tablist"
+        aria-labelledby="scan-results-header"
+      >
+        ${tabs.map(
```

Comment:
Not for now, but we should probably add a generic tab component at some point, so we don't keep duplicating this logic

Reply:
yes, agree